R/surv.b.R

```diff
@@ -40,150 +52,237 @@ survClass <- R6::R6Class(
-                object=state,
-                fun=fun,
-                xlab='Elapsed',
-                ylab=ylab,
-                ylim=ylim,
-                surv.size = 1,
-                censor.size = 8,
-                censor.alpha = 0.8,
-                conf.int=ci,
-                censor=cens)
+            s <- survival::survfit(survival::Surv(times, status) ~ group, data=tmpDat)
+            if (!is.null(names(s$strata))) names(s$strata) <- gsub("group=", "", names(s$strata))
+
+            conf.int <- self$options$ci
+            censYN <- self$options$cens
+            risktableYN <- self$options$risktable
+            legend <- "top"
+            legendtitle <- self$options$groups
+            if (is.null(self$options$groups)) {
+                legend <-"none"
+                legendtitle <- NULL
+                
+            } 
+            xlab <- "Time"
+            if (!self$options$timeunits=="") xlab <- paste("Time (",self$options$timeunits,")",sep="")
+            survmedianline <-"none"
+            if (self$options$plotmedian) survmedianline <- "hv"
             
-            plot <- plot + ggtheme
             
-            print(plot)
+            if (identical(image, self$results$sc)) {
+                func <- NULL
+                ylab <- "Survival Probabilities"
+            } else {
+                if (!self$options$logscale) {
+                    func <- "cumhaz"
+                    ylab <- "Cumulative Hazards"
+                    conf.int <- FALSE
+                    censYN <- FALSE
+                    risktableYN <- FALSE
+                    survmedianline <-"none"
+                    
+                    
+                } else {
+                    func <- function(x) log(-log(x))
+                    ylab <- "Cumulative Hazards (log scale)"
+                    conf.int <- FALSE
+                    censYN <- FALSE
+                    risktableYN <- FALSE
+                    survmedianline <-"none"
+                    
+                    
+                }
+                    
+                } 
+                    
+            p <- survminer::ggsurvplot(s,
+                                       data = tmpDat,
+                                       xlab=xlab,
+                                       ylab = ylab,
+                                       conf.int = conf.int,
+                                       legend=legend,
+                                       legend.title=legendtitle,
+                                       censor = censYN,
+                                       fun = func,
+                                       censor.shape = "|",
+                                       censor.size = 6,
+                                       risk.table = risktableYN,
+                                       surv.median.line = survmedianline
+                                       )
             
+            print(p)
             return(TRUE)
```

Comment:
rather than printing `p` and returning `TRUE`, you can just return `p` ... i've been meaning to update documentation on this one. returning `p` is better.

Reply:
I made the change :-)